compiler/luci/pass/src/QuantizedModelVerifyPass.cpp

```diff
@@ -0,0 +1,455 @@
+      RETURN_FALSE_UNLESS(has_type(node->y(), Type::U8))
+    }
+    else if (_type == Type::S16)
+    {
+      RETURN_FALSE_UNLESS(has_type(node, Type::S16))
+      RETURN_FALSE_UNLESS(has_type(node->x(), Type::S16))
+      RETURN_FALSE_UNLESS(has_type(node->y(), Type::S16))
+    }
+    return true;
+  }
+
+  bool visit(const luci::CircleAveragePool2D *node)
+  {
+    if (_type == Type::U8)
+    {
+      RETURN_FALSE_UNLESS(has_type(node, Type::U8))
+      RETURN_FALSE_UNLESS(has_type(node->value(), Type::U8))
+    }
+    else if (_type == Type::S16)
+    {
+      RETURN_FALSE_UNLESS(has_type(node, Type::S16))
+      RETURN_FALSE_UNLESS(has_type(node->value(), Type::S16))
+    }
+    return true;
+  }
+
+  bool visit(const luci::CircleMaxPool2D *node)
+  {
+    if (_type == Type::U8)
+    {
+      RETURN_FALSE_UNLESS(has_type(node, Type::U8))
+      RETURN_FALSE_UNLESS(has_type(node->value(), Type::U8))
+    }
+    else if (_type == Type::S16)
+    {
+      RETURN_FALSE_UNLESS(has_type(node, Type::S16))
+      RETURN_FALSE_UNLESS(has_type(node->value(), Type::S16))
+    }
+    return true;
+  }
+
+  bool visit(const luci::CircleMean *node)
+  {
+    if (_type == Type::U8)
+    {
+      RETURN_FALSE_UNLESS(has_type(node, Type::U8))
+      RETURN_FALSE_UNLESS(has_type(node->input(), Type::U8))
+      RETURN_FALSE_UNLESS(has_type(node->reduction_indices(), Type::S32))
+    }
+    else if (_type == Type::S16)
+    {
+      RETURN_FALSE_UNLESS(has_type(node, Type::S16))
+      RETURN_FALSE_UNLESS(has_type(node->input(), Type::S16))
+      RETURN_FALSE_UNLESS(has_type(node->reduction_indices(), Type::S32))
+    }
+    return true;
+  }
+
+  bool visit(const luci::CircleMul *node)
+  {
+    if (_type == Type::U8)
+    {
+      RETURN_FALSE_UNLESS(has_type(node, Type::U8))
+      RETURN_FALSE_UNLESS(has_type(node->x(), Type::U8))
+      RETURN_FALSE_UNLESS(has_type(node->y(), Type::U8))
+    }
+    else if (_type == Type::S16)
+    {
+      RETURN_FALSE_UNLESS(has_type(node, Type::S16))
+      RETURN_FALSE_UNLESS(has_type(node->x(), Type::S16))
+      RETURN_FALSE_UNLESS(has_type(node->y(), Type::S16))
+    }
+    return true;
+  }
+
+  bool visit(const luci::CircleRelu *node)
+  {
+    if (_type == Type::U8)
+    {
+      RETURN_FALSE_UNLESS(has_type(node, Type::U8))
+      RETURN_FALSE_UNLESS(has_type(node->features(), Type::U8))
+    }
+    else if (_type == Type::S16)
+    {
+      RETURN_FALSE_UNLESS(has_type(node, Type::S16))
+      RETURN_FALSE_UNLESS(has_type(node->features(), Type::S16))
+    }
+    return true;
+  }
+
+  bool visit(const luci::CircleNode *node)
+  {
+    if (_type == Type::U8)
+    {
+      RETURN_FALSE_UNLESS(has_type(node, Type::U8))
+    }
+    else if (_type == Type::S16)
+    {
+      RETURN_FALSE_UNLESS(has_type(node, Type::S16))
+    }
+    return true;
+  }
+
+  // TODO: Implement more Ops
+};
+
+} // namespace
+
+namespace luci
+{
+
+bool QuantizedModelVerifyPass::run(loco::Graph *g)
```

Comment:
(optiona) instead of using `Pass` in suffix, how about `QuantizedModelVerifier` ?
as I understand, this uses `Pass` framework but doesn't work like other transoformation passes.
To be strict, it would be bette not to inherit from `logo::Pass`.

Reply:
I agree that `Pass` is not necessary here. I'll remove `logo::Pass`. Thanks.